portfolio/src/main/webapp/script.js

```diff
@@ -38,3 +38,57 @@ function myFunction() {
     x.className = "nav";
   }
 }
+
+function getHello() {
+  console.log('Fetching a random message.');
+  
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+*/
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+	
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addToDom);
+}
+
+/** Adds a random message to the DOM. */
+function addToDom(message) {
+  console.log('Adding quote to dom: ' + message);
+  const messageContainer = document.getElementById('message-container');
+  messageContainer.innerText = message;
+}
+
+// async version of the get HellowMessage
+async function getHelloAwait() {
+  const response = await fetch('/data');
+  const message = await response.text();
+  document.getElementById('message-container').innerText = message;
+}
+
+// parsing & fetching json messages
+async function getComment() {
+  const response = await fetch('/data');
+  const comments = await response.json();
+  const element = document.getElementById('comment-container');
+  
+  comments.forEach((line) => {
+    const newComment = document.createElement('li');
+    newComment.innerText = line.date + " : " + line.name + " said " 
```

Comment:
There are multiple better ways to do this. Probably [HTML Templates](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) are the nicest way.

Reply:
Thanks. I will look into it.